src/libstd/path.rs

```diff
@@ -1520,6 +1532,10 @@ impl<'a> From<Cow<'a, Path>> for PathBuf {
 
 #[stable(feature = "shared_from_slice2", since = "1.24.0")]
 impl From<PathBuf> for Arc<Path> {
+    /// Converts a `PathBuf` into a `Arc<Path>`.
+    /// This conversion happens in place.
+    /// This conversion does not allocate memory.
+    /// This function is unsafe. Data can't be moved from this reference.
```

Comment:
How is this function unsafe? It isn't marked as such.

Reply:
Hm, I was thinking because of https://github.com/rust-lang/rust/pull/51753/files#diff-120c5906285d41e976dc4176265d7cbaR1542. It calls internally an unsafe function. But I guess if the calling method is safe, then it doesn't matter if it calls `unsafe` internally?

Reply:
Yes, Rust practice is to assume that the unsafety is contained in the block. Calling the function unsafe declares that people need to fulfil certain conditions before calling the function to ensure it can be safely called.

The reasoning behind that is that otherwise, everything would be transitively unsafe.